src/UI/PlayerSetupColorFactionController.java

```diff
@@ -31,10 +45,26 @@ public boolean handleInput(InputAction action)
     switch(action)
     {
       case SELECT:
-        // Apply change and return control.
-        myPlayerInfo.coList.get(0).color = colorSelector.getSelectionNormalized();
-        myPlayerInfo.coList.get(0).faction = factionSelector.getSelectionNormalized();
-        done = true;
+        colorPicks[tagIndex]   = colorSelector  .getSelectionNormalized();
+        factionPicks[tagIndex] = factionSelector.getSelectionNormalized();
+        ++tagIndex;
+        // If we have more tag COs, stick around and handle those...
+        if( tagIndex < colorPicks.length )
+        {
+          // Navigate the cursor to the current value of that index's position
+          // Is this good UX? Nobody knows!
```

Comment:
It's as good a starting point as any. I'm not sure what I'd do differently on this one.